src/java.base/share/classes/sun/security/ec/ECPrivateKeyImpl.java

```diff
@@ -253,4 +240,41 @@ private void readObject(ObjectInputStream stream)
         throw new InvalidObjectException(
                 "ECPrivateKeyImpl keys are not directly deserializable");
     }
+
+    // Parse the SEC1v2 encoding to extract public key, if available.
+    public static BitArray parsePublicBits(byte[] privateBytes) {
+        DerValue seq = null;
+        try {
+            seq = new DerValue(privateBytes);
+            if (seq.tag == DerValue.tag_Sequence) {
+                int version = seq.data.getInteger();
+                if (version == 1) { // EC
+                    byte[] oct = seq.data.getOctetString();  // private key
```

Comment:
If you call `seq.data.getDerValue()` then the return value uses the original array with an offset and a length. So there is no need to call `Arrays.fill` next.